docs: add release notes for 0.31.0 (#9159)
determined-ai/determined: docs/release-notes.rst

 Release Notes
###############

**************
 Version 0.31
**************

Version 0.31.0
==============

**Release Date:** April 15, 2024

**New Features**

-  API: Add a new API endpoint, ``/health``, that provides information about the status of
   Determined's connections to the database, Kubernetes API server, and Slurm launcher integration.

   See the REST API documentation for more information about the endpoint.

> 

Reviewer: Visit the :ref:`rest-api` documentation for more information about this endpoint.

Reviewer: ":ref:`rest-api`
documentation"

Reviewer: Did this actually get added to the REST API docs page?

Author: `:ref:rest-api` goes to http://latest-main.determined.ai:8080/docs/reference/rest-api.html#rest-api-reference

   ``retention_policy.log_retention_days`` config option. See :ref:`master-config-reference` and
   :ref:`experiment-config-reference` for more details.

-  CLI: Add commands, ``det e set log-retention <exp-id>`` and ``det t set log-retention
   <trial-id>``, to allow the user to set the length of log retention for experiments and trials.
   Both commands can specify a length in days with the arguments ``--days <number of days>``. The
   number of days must be between -1 and 32767, where -1 retains logs forever. ``--forever`` is
   equivalent to ``--days -1``.

-  WebUI: Add support for retaining logs for multiple experiments by selecting experiments from the
   experiment list page and choosing **Retain Logs** from **Actions**. Users can then input the
   desired number of days for log retention or select the "Forever" checkbox for indefinite log
   retention. The number of days must be between -1 and 32767, where -1 retains logs forever.

   There is a new column on the trial list page, "Log Retention Days", that displays the number of
   days for which logs will be retained for each trial after creation.

> -  Master config: Ad a new field to task container defaults named ``startup_hook`` that allows for

Reviewer: "Ad" should be "Add"

Author: fixed

 Release Notes
###############

**************
 Version 0.31
**************

Version 0.31.0
==============

**Release Date:** April 15, 2024

**New Features**

> -  API: Add a new API ``/health`` that has information about the status of Determined's database
>    connection, connection to Kubernetes API server, and connection to Slurm launcher integration.

Reviewer 2: ```suggestion
-  API: Add a new API endpoint, ``/health``, that provides information about the status of
   Determined's connections to the database, Kubernetes API server, and Slurm launcher integration.
```

>    The REST API documentation can be viewed for more information about the endpoint.

Reviewer 2: ```suggestion
   See the REST API documentation for more information about the endpoint.
```

> -  Logging: Add a configuration option, ``logging_retention``, to the server configuration file with
>    options for scheduling (``schedule``) log cleanup, and for selecting the number of ``days`` to
>    retain logs. Experiments can override the default log retention settings provided by the server
>    by specifying ``log_retention_days`` in the experiment configuration. Valid values for retention
>    days range from ``-1`` to ``32767``, and schedules must adhere to a valid cron expression or
>    duration format. If retention days is set to ``-1``, logs will be retained indefinitely.
>    Conversely, setting retention days to 0 will result in logs being deleted during the next
>    scheduled log cleanup. Additionally, administrators can manually initiate log retention cleanup
>    using the ``det task cleanup-logs command``.

Reviewer 2: ```suggestion
-  Logging: Add a ``retention_policy`` section to the master config file for specifying the default
   log retention policy. Experiments can override the default log retention settings with the
   ``retention_policy.log_retention_days`` config option. See :ref:`master-config-reference` and
   :ref:`experiment-config-reference` for more details.
```

> -  CLI: Add commands, ``det e set log-retention <exp-id>`` and ``det t set log-retention
>    <trial-id>`` to allow the user to set the log retention days for experiments and trials. These
>    commands take the required argument ``--days`` and the optional argument ``forever``. ``--days``
>    sets the number of days to retain the logs for, from the time of creation and ``--forever``
>    retains logs forever. The allowed range for ``--days`` is between ``-1`` and ``32767``, where -1
>    retains logs forever.

Reviewer 2: ```suggestion
-  CLI: Add commands, ``det e set log-retention <exp-id>`` and ``det t set log-retention
   <trial-id>``, to allow the user to set the length of log retention for experiments and trials.
   Both commands can specify a length in days with the arguments ``--days <number of days>``. The
   number of days must be between -1 and 32767, where -1 retains logs forever. ``--forever`` is
   equivalent to ``--days -1``.
```

-  WebUI: Add support for retaining logs for multiple experiments by selecting experiments from the

>    Experiment List page and choosing **Retain Logs** from **Actions**. Users can then input the

Reviewer 2: ```suggestion
   experiment list page and choosing **Retain Logs** from **Actions**. Users can then input the
```

   desired number of days for log retention or select the "Forever" checkbox for indefinite log

>    retention. The allowed range for the number of days is between ``-1`` and ``32767``, where -1
>    retains logs forever. Add a new column, ''Log Retention Days'', to the Trial List page. ''Log
>    Retention Days'' displays the duration logs will be retained for each trial from creation.

Reviewer 2: ```suggestion
   retention. The number of days must be between -1 and 32767, where -1 retains logs forever.

   There is a new column on the trial list page, "Log Retention Days", that displays the number of
   days for which logs will be retained for each trial after creation.
```

> -  Master Config: Added a new field to task container defaults named ``startup_hook`` which allows
>    for the specification of an inline script to be executed after task setup.

Reviewer 2: ```suggestion
-  Master config: Ad a new field to task container defaults named ``startup_hook`` that allows for
   the specification of an inline script to be executed after task setup.
```

**Improvements**

-  CLI: The ``--add-tag`` flag to ``det deploy aws up`` will now apply tags to dynamic agents
   launched.

**Bug Fixes**

> -  API: Fix a bug where calling ``det job update`` could prevent jobs from being scheduled and ``det
>    job ls`` to hang.

Reviewer 2: ```suggestion
-  API: Fix a bug where calling ``det job update`` could prevent jobs from being scheduled and cause
   ``det job ls`` to hang.
```

**Security Fixes**

-  Helm: When deploying a new cluster with Helm, configuring an initial password for the "admin" and
   "determined" users is required and is no longer a separate step. To specify an initial password

>    for these users, visit the helm/charts/determined/values.yaml file and use either
>    initialUserPassword (preferred) or defaultPassword (deprecated). For reference, visit
>    :ref:`helm-config-reference`.

Reviewer 2: ```suggestion
   for these users, set either ``initialUserPassword`` (preferred) or ``defaultPassword``
   (deprecated) in the ``helm/charts/determined/values.yaml`` file. For reference, see
   :ref:`helm-config-reference`.
```

 Release Notes
###############

**************
 Version 0.31
**************

Version 0.31.0
==============

**Release Date:** April 15, 2024

**Breaking Changes**

-  SAML: The underlying SAML implementation has been updated to use a newer, more maintained
   library. As a result, the master config no longer accepts the `idp_cert_path` field and now

>    requires the `idp_metadata_url` field when using SAML.

Author: @tara-det-ai @dzhu 
@eecsliu made this change because this release note was not in OSS repo, but in ee repo. could you review this as well?

Reviewer: i belive any docs changes in the ee repo just get overwritten at release time

Reviewer: @eecsliu replied that these were already moved to OSS docs, i'll mark as resolved

>    library. As a result, the master config no longer accepts the `idp_cert_path` field and now
>    requires the `idp_metadata_url` field when using SAML.

Reviewer 2: ```suggestion
   library. As a result, the master config no longer accepts the ``idp_cert_path`` field and now
   requires the ``idp_metadata_url`` field when using SAML.
```

**New Features**

-  API: Add a new API endpoint, ``/health``, that provides information about the status of
   Determined's connections to the database, Kubernetes API server, and Slurm launcher integration.

   Visit the :ref:`rest-api` documentation for more information about this endpoint.

-  Logging: Add a ``retention_policy`` section to the master config file for specifying the default
   log retention policy. Experiments can override the default log retention settings with the
   ``retention_policy.log_retention_days`` config option. See :ref:`master-config-reference` and
   :ref:`experiment-config-reference` for more details.

> -  CLI: Add commands, ``det e set log-retention <exp-id>`` and ``det t set log-retention

Reviewer 3: CLI: Add commands, ``det e set log-retention <exp-id>`` and ``det t set log-retention <trial-id>`` to allow the user to set the log retention days for experiments and trials. These commands take the required argument ``--days`` and the optional argument ``forever``. ``--days`` sets the number of days to retain the logs for, from the time the task ended and ``--forever`` retains logs forever. The allowed range for ``--days`` is between ``-1`` and ``32767``, where -1 retains logs forever.

 Release Notes
###############

**************
 Version 0.31
**************

Version 0.31.0
==============

> **Release Date:** April 15, 2024

Author: # REMINDER FOR MYSELF: **_CHANGE DATE BEFORE MERGE_**

 Release Notes
###############

**************
 Version 0.31
**************

Version 0.31.0
==============

**Release Date:** April 12, 2024

**New Features**

-  API: Add a new API ``/health`` that has information about the status of Determined's database
   connection, connection to Kubernetes API server, and connection to Slurm launcher integration.

   The REST API documentation can be viewed for more information about the endpoint.

> -  Logging: Add a configuration option, ``logging_retention``, to the server configuration file with

Reviewer 3: Logging: Add a configuration option, ``retention_policy``, to the server configuration file with options for scheduling (``schedule``) log cleanup, and for selecting the number of ``log_retention_days`` to retain logs. Experiments can override the default log retention settings provided by the server by specifying ``log_retention_days`` in the experiment configuration. Valid values for retention days range from ``-1`` to ``32767``, and schedules must adhere to a valid cron expression or duration format. If ``log_retention_days`` is set to ``-1``, logs will be retained indefinitely. Conversely, setting ``log_retention_days`` to 0 will result in logs being deleted during the next scheduled log cleanup. Additionally, administrators can manually initiate log retention cleanup using the ``det task cleanup-logs`` command.
